### DISCUSSION TITLE
Memory Leaks

### DISCUSSION THREAD

**Discussion by kirankumar-ravuri** [baresip/re#753] [General]
Hi All,

I am trying to use libre in MacOS and trying just registration and unregister APIs. I followed the example code in http://creytiv.com/pub/redemo-0.5.0.tar.gz. 
I am seeing below errors at the end after calling re_cancel. 

main: re_tmrl_get: re not ready
mem: Memory leaks (66):

Would someone like to point me to where I should be looking into to fix this?

BRs
Kiran

**Comment by sreimers** [comment]
redemo is maybe not compatible with `baresip/re` and outdated. You can look at https://github.com/baresip/re/tree/main/test and https://github.com/baresip/baresip for usage examples. Or you can use baresip as a library: https://github.com/baresip/baresip/wiki/Using-baresip-as-a-library

**Comment by kirankumar-ravuri** [comment]
@sreimers Thanks a lot. Tests helped in clearing out the leaks.
